src/Cli/dotnet/Telemetry/Telemetry.cs

```diff
@@ -112,7 +112,7 @@ public void Flush()
                 return;
             }
 
-            _trackEventTask.Wait();
+            Task.Run(() => _trackEventTask.Wait());
```

Comment:
Since this is in the `Flush` method, this feels incorrect. Since `Task.Run`'s task isn't waited, the `Flush` method will return before `_trackEventTask` has finished, and therefore the telemetry might not yet have been flushed. If the process terminates before the `_trackEventTask` completes, there will be data loss (unless the telemetry is backed by a file that is read on next process startup?)